client-tools/Shell/ClientFeature.cpp

```diff
@@ -396,7 +398,19 @@ std::unique_ptr<httpclient::SimpleHttpClient> ClientFeature::createHttpClient(
                                        _connectionTimeout, _retries,
                                        _sslProtocol));
 
```

Comment:
Maybe add an assertion that the connection is not nullptr to create a client using it in the following line?